modules/internal/ChapelBase.chpl

```diff
@@ -153,12 +153,17 @@ module ChapelBase {
   inline proc ==(param a: bool, param b: bool) param return __primitive("==", a, b);
   inline proc ==(param a: int(?w), param b: int(w)) param return __primitive("==", a, b);
   inline proc ==(param a: uint(?w), param b: uint(w)) param return __primitive("==", a, b);
+  //
   inline proc ==(param a: enumerated, param b: enumerated) param where (a.type == b.type) return __primitive("==", a, b);
+  //
+  // NOTE: Only '==' is implemented in the compiler on param enums
+  // (and only only on param enums of the same type), so all other
```

Comment:
"only" repeats twice.
Also, I did not see where the compiler checks whether the two enums are of the same type.
Adjust comment? Adjust the compiler?

Reply:
The compiler doesn't check, but assumes they're the same.  I've updated the comment to clarify that the where clause here is what's keeping things safe.